test/DynamoCoreTests/DynamoDefects.cs

```diff
@@ -472,5 +472,24 @@ public void EqualEqualTest_ForStrings_Defect6694()
             // Same Stinrg input
             AssertPreviewValue("56f3c0fd-d39c-46cb-a4ea-4f266f7a9fce", true);
         }
+
+        [Test, Category("RegressionTests")]
+        public void CyclicDependency_Defect8827()
+        {
+            string openPath = Path.Combine(TestDirectory,
+                                        @"core\DynamoDefects\Defect_MAGN_8827.dyn");
+            RunModel(openPath);
+            // check for number of Nodes and Connectors
+            Assert.AreEqual(2, CurrentDynamoModel.CurrentWorkspace.Nodes.Count());
+            Assert.AreEqual(2, CurrentDynamoModel.CurrentWorkspace.Connectors.Count());
+
+            var node1 = CurrentDynamoModel.CurrentWorkspace.NodeFromWorkspace
+                ("385281c8-3177-4359-a348-a3084e16a41a");
+            var node2 = CurrentDynamoModel.CurrentWorkspace.NodeFromWorkspace
+                ("1ad8632e-7ddc-4cc7-bfa5-58cb899a5ddf");
+
+            Assert.IsTrue(node1.ToolTipText.Contains("cyclic"));
```

Comment:
You could check for the whole message here: "A cyclic dependency exists between two variables", could you not?